File: benchmarks/python/xinsheyangn2fcn.py
import numpy as np

def xinsheyangn2fcn(x: np.ndarray) -> np.ndarray:
    """
    Computes the value of the Xin-She Yang N. 2 function.
    
    Parameters:
    -----------
    x (numpy.ndarray): input matrix of size M-by-N
    
    Returns:
    --------
    numpy.ndarray: vetor SCORES of size M-by-1 in which each row contains the function value for the corresponding row of X
    
    Author: Mazhar Ansari Ardeh
    Please forward any comments or bug reports to mazhar.ansari.ardeh at
    Google's e-mail service or feel free to kindly modify the repository.
    """
    scores = np.sum(np.abs(x), axis=1) * np.exp(-np.sum(np.sin(x ** 2), axis=1))
    return scores

File: src/matlab/xinsheyangn2fcn.m
% Computes the value of the Xin-She Yang N. 2 function.
% SCORES = XINSHEYANGN2FCN(X) computes the value of the Xin-She Yang N. 2
% function at point X. XINSHEYANGN2FCN accepts a matrix of size M-by-N and
% returns a vetor SCORES of size M-by-1 in which each row contains the
% function value for the corresponding row of X.
% For more information, please visit:
% benchmarkfcns.info/doc/xinsheyangn2fcn
%
% Author: Mazhar Ansari Ardeh
% Please forward any comments or bug reports to mazhar.ansari.ardeh at
% Google's e-mail service or feel free to kindly modify the repository.
function scores = xinsheyangn2fcn(x)
     scores = sum(abs(x), 2) .* exp(-sum(sin(x .^2), 2));
end
